plugins/pulp_rpm/plugins/importers/yum/pulp_solv.py

```diff
@@ -0,0 +1,408 @@
+    solvable object; these are created in the pool object:
+
+        dependency = pool.Dep('foo')
+
+    The relationship to the solvable is tracked by a Rel pool object:
+
+        relationship = pool.Rel(solv.REL_AND, pool.Dep(evr))
+
+    where the evr is the 'epoch:version-release' string. The relationship
+    is then recorded on the solvable explicitly by:
+
+        solvable.add_deparray(solv.SOLVABLE_PROVIDES, relationship)
+
+    If no explict relationship is provided in the flags attribute,
+    the dependency can be used directly:
+
+        solvable.add_deparray(solv.SOLVABLE_PROVIDES, dependency)
+
+    :param solvable: a libsolv solvable object
+    :type solvable: a libsolv solvable
+    :param unit: the content unit to get the dependencies from
+    :type unit: an object or a dictionary
+    :returns: None
+    """
+    # e.g SOLVABLE_PROVIDES, SOLVABLE_REQUIRES...
+    keyname = dependency_key or getattr(solv, 'SOLVABLE_{}'.format(attr_name.upper()))
+    pool = solvable.repo.pool
+    if unit_name.startswith('('):
+        # the Rich/Boolean dependencies have just the 'name' attribute
+        # this is always in the form: '(foo >= 1.2 with foo < 2.0)'
+        dep = pool.parserpmrichdep(unit_name)
+    else:
+        # generic dependencies provide at least a solvable name
+        dep = pool.Dep(unit_name)
+        if unit_flags:
+            # in case the flags unit attribute is populated, use it as
+            # a solv.Rel object to denote solvable--dependency
+            # relationship dependency in this case is a relationship
+            # towards the dependency made from the 'flags', e.g:
+            # solv.REL_EQ, and the evr fields
+            if unit_flags == 'EQ':
+                rel_flags = solv.REL_EQ
+            elif unit_flags == 'LT':
+                rel_flags = solv.REL_LT
+            elif unit_flags == 'GT':
+                rel_flags = solv.REL_GT
+            elif unit_flags == 'LE':
+                rel_flags = solv.REL_EQ | solv.REL_LT
+            elif unit_flags == 'GE':
+                rel_flags = solv.REL_EQ | solv.REL_GT
+            else:
+                raise ValueError('Unsupported dependency flags %s' % unit_flags)
+            dep = dep.Rel(rel_flags, pool.Dep(unit_evr))
+        # register the constructed solvable dependency
+        solvable.add_deparray(keyname, dep)
+
+
+def rpm_dependency_attribute_factory(attribute_name, dependency_key=None):
+    return repeated_attr_conversion(attr_conversion(attribute_name, default=[])())(
+        lambda solvable, unit: rpm_dependency_conversion(
+            solvable, unit, attribute_name, dependency_key=dependency_key
+        ))
+
+
+def rpm_filelist_conversion(solvable, unit):
+    """A specific, rpm-unit-type filelist attribute conversion."""
+    repodata = solvable.repo.first_repodata()
+    unit_files = unit.get('files', {}).get('file', [])
+    unit_files.extend(unit.get('files', {}).get('dir', []))
```

Comment:
both the `rpm.files['file']` and `rpm.files['dir']` have to be processed the same `os.path.dirname(); os.path.basename()` way for `libsolv` to understand.